src/main/java/frc/team2412/robot/Controls.java

```diff
@@ -95,10 +101,12 @@ private void bindDrivebaseControls() {
 								driveController::getLeftY,
 								driveController::getLeftX,
 								() -> Rotation2d.fromRotations(driveController.getRightX())));
-		driveController.start().onTrue(new InstantCommand(s.drivebaseSubsystem::resetGyro));
 		driveController.rightStick().onTrue(new InstantCommand(s.drivebaseSubsystem::toggleXWheels));
 	}
 
+	public void bindLimelightControls() {
+		getWithinDistanceTrigger.onTrue(s.limelightSubsystem.getWithinDistance(s.drivebaseSubsystem));
```

Comment:
```suggestion
		getWithinDistanceTrigger.onTrue(
		        Commands.deferred(
		                () -> s.limelightSubsystem.getWithinDistance(s.drivebaseSubsystem),
		                Set.of(s.limelightSubsystem, s.drivebaseSubsystem)));
```
Otherwise, `getWithinDistance()` would only be called once, so `hasTargets()` would only be checked once (at the start of robot code). Alternately, make the supplier used by `getWithinDistance()` check `hasTargets()` (a ternary operator would be useful here), so that `hasTargets()` will be repeatedly checked.

Reply:
Outdated because of new push and logic restructuring